mip/constraint.go

```diff
@@ -0,0 +1,63 @@
+package mip
+
+// Sense defines the constraint operator between the left-hand-side
+// and the right-hand-side.
+type Sense int64
+
+// Sense of a Constraint.
+const (
+	// LessThanOrEqual used to define a less than equal constraint
+	// 		c, _ := d.NewConstraint(mip.LessThanOrEqual, 123.4)
+	//
+	// 		c.NewTerm(1.0, x)  	 // results in 1.0 * x <= 123.4 in solver
+	LessThanOrEqual Sense = iota
+	// Equal used to define a equal constraint
+	// 		c, _ := d.NewConstraint(mip.Equal, 123.4)
+	//
+	// 		c.NewTerm(1.0, x)  	 // results in 1.0 * x = 123.4 in solver
+	Equal
+	// GreaterThanOrEqual used to define a greater or equal constraint
+	// 		c, _ := d.NewConstraint(mip.	GreaterThanOrEqual, 123.4)
+	//
+	// 		c.NewTerm(1.0, x)  	 // results in 1.0 * x >= 123.4 in solver
+	GreaterThanOrEqual
+)
+
+// Constraint specifies a relation between variable a solution has to comply
+// with. A constraint consists out of terms, a sense and a right hand side.
+//
+// For example:
+//
+//	2.5 * x + 3.5 * y < 10.0
```

Comment:
```suggestion
//	2.5 * x + 3.5 * y <= 10.0
```
Maybe change the example to less than or equal, since we don't support the strictly less constraint, right?